core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-user/dot-toolbar-user.component.ts

```diff
@@ -17,34 +20,79 @@ import { IframeOverlayService } from '../../../_common/iframe/service/iframe-ove
     templateUrl: 'dot-toolbar-user.component.html'
 })
-export class DotToolbarUserComponent implements OnInit {
+export class DotToolbarUserComponent implements OnInit, OnDestroy {
     @ViewChild(DotDropdownComponent) dropdown: DotDropdownComponent;
+
+    items: MenuItem[];
+
     auth: Auth;
 
     showLoginAs = false;
     showMyAccount = false;
 
     logoutUrl = `${LOGOUT_URL}?r=${new Date().getTime()}`;
 
-    haveLoginAsPermission$: Observable<boolean>;
+    private destroy$: Subject<boolean> = new Subject<boolean>();
 
     constructor(
         @Inject(LOCATION_TOKEN) private location: Location,
         private loggerService: LoggerService,
         private loginService: LoginService,
         public iframeOverlayService: IframeOverlayService,
         private dotNavigationService: DotNavigationService,
-        private dotRouterService: DotRouterService
+        private dotRouterService: DotRouterService,
+        private dotMessageService: DotMessageService
     ) {}
 
     ngOnInit(): void {
         this.loginService.watchUser((auth: Auth) => {
             this.auth = auth;
         });
 
-        this.haveLoginAsPermission$ = this.loginService
+        this.loginService
             .getCurrentUser()
-            .pipe(map((res: CurrentUser) => res.loginAs));
+            .pipe(
+                map((res: CurrentUser) => res.loginAs),
+                takeUntil(this.destroy$)
+            )
+            .subscribe((result) => {
+                this.items = [
+                    {
+                        id: 'dot-toolbar-user-link-my-account',
+                        label: this.dotMessageService.get('my-account'),
+                        icon: 'pi pi-user-edit',
+                        visible: !this.auth.loginAsUser,
+                        command: () => this.toggleMyAccount()
+                    },
+                    {
+                        id: 'dot-toolbar-user-link-login-as',
+                        label: this.dotMessageService.get('login-as'),
+                        icon: 'pi pi-sort-alt',
+                        visible: !this.auth.loginAsUser && result,
+                        command: () => this.tooggleLoginAs()
+                    },
+                    { separator: true },
+                    {
+                        id: 'dot-toolbar-user-link-logout',
+                        label: this.dotMessageService.get('Logout'),
+                        icon: 'pi pi-sign-out',
+                        visible: !this.auth.loginAsUser,
+                        url: this.logoutUrl
+                    },
+                    {
+                        id: 'dot-toolbar-user-link-logout-as',
+                        label: this.dotMessageService.get('logout-as'),
+                        icon: 'pi pi-sign-out',
+                        visible: !!this.auth.loginAsUser,
+                        command: (event) => this.logoutAs(event.originalEvent)
+                    }
+                ];
+            });
```

Comment:
You can use async pipe here, I don't have the code in my mind to do a suggestions, but ping me and I will help you.

Reply:
I think I had an idea. I'll ping you if needed